compute_endpoint/tests/unit/test_endpoint_config.py

```diff
@@ -31,3 +39,24 @@ def test_config_model_tuple_conversions(config_dict: dict, data: t.Tuple[str, t.
     config_dict["engine"][field] = 50000
     with pytest.raises(ValueError):
         ConfigModel(**config_dict)
+
+
+valid_ports = get_args(AMQP_PORT_TYPE)
+
+
+def test_invalid_amqp_port(config_dict):
+    ports_to_sample = [p for p in range(1, 65536) if p not in valid_ports]
+    # just do 20 samples for speed
+    for _ in range(20):
+        port = random.choice(ports_to_sample)
+        # in expected cases, swallowed by pytest. in exceptional cases,
+        # shows what port got through
+        print(port)
+
+        with pytest.raises(ValueError):
+            ConfigModel(amqp_port=port, **config_dict)
```

Comment:
I recommend this syntax, which keeps the looping and sampling outside of the main test body; pytest will also track any failing port numbers without needing a `print()`.

```suggestion
valid_ports = get_args(AMQP_PORT_TYPE)
invalid_ports = [p for p in range(1, 65536) if p not in valid_ports]


@pytest.mark.parametrize("port", random.sample(invalid_ports, k=20))
def test_invalid_amqp_port(config_dict, port):
        with pytest.raises(ValueError):
            ConfigModel(amqp_port=port, **config_dict)
```

Reply:
Much nicer syntax 👍